### ISSUE TITLE
Randomness?

### ISSUE THREAD

**Issue by rightx2** [catboost/catboost#1066]
Problem: Randomness
catboost version: 0.17.5

I have set of grid params (total 10 sets, consisting of loss function, eval metric, and version of my own feature data etc, not argument for a model.) and run 2 SAME experiments with this model:
```
model = catboost.CatBoostRegressor(
    iterations=100,
    task_type="GPU",
    loss_function=loss,
    verbose=0,
    colsample_bylevel=1,
    random_strength=0,
)
a = model.fit(
    train_x, train_y,
    use_best_model=True,
    eval_set=[(valid_x, valid_y,)],
    verbose=0,
    early_stopping_rounds=early_stopping_rounds,
)
```
As you can see here, I fixed randomness param of catboost.

But with the same param set(like 2~3), some of scores and metrics show different values in both SAME experiments even though they used PARAM SET, SAME DATA.

Anything I missed for fixed randomness?

**Comment by annaveronika** [comment]
GPU training is not deterministic. It's stated in the docs here: https://catboost.ai/docs/concepts/faq.html#differencies-between-cpu-and-gpu

**Comment by annaveronika** [comment]
I hope it answers your question! Closing the issue.
